src/content/blog/astro-4120.mdx

```diff
@@ -0,0 +1,140 @@
+---
+title: "Astro 4.12: Server Islands"
+description: Astro 4.12 is now available! This release includes a first look at Server Islands, improvements to pagination and syntax highlighting, and more.
+publishDate: "July 18, 2024"
+authors:
+  - erika
+  - matthew
+coverImage: "/src/content/blog/_images/astro-4120/header_astro_4_12.webp"
+socialImage: "/src/content/blog/_images/astro-4120/og_image_4_12.webp"
+lang: "en"
+---
+
+import BlogContentImage from "/src/components/BlogContentImage.astro";
+import BlogSummary from "/src/components/BlogSummary.astro";
+import islandDiag from "/src/content/blog/_images/future-of-astro-server-islands/dark-mode-server-islands-diag.webp";
+
+<BlogSummary>
+  Astro 4.12 is out now! This release includes a first look at [Server Islands](https://astro.build/blog/future-of-astro-server-islands/), as well as improvements to pagination and syntax highlighting, and more.
+</BlogSummary>
+
+This release includes the following highlights:
+
+- [**Experimental: Server Islands**](#experimental-server-islands)
+- [**Added `first` and `last` URLs to pagination data**](#added-first-and-last-urls-to-pagination-data)
+- [**Added support for the Shiki's `defaultColor` option**](#added-support-for-the-shikis-defaultcolor-option)
+- [**New `inferRemoteSize` function**](#new-inferremotesize-function)
+
+To upgrade an existing project, use the automated `@astrojs/upgrade` CLI tool. Alternatively, upgrade manually by running the upgrade command for your package manager:
+
+```sh
+# Recommended:
+npx @astrojs/upgrade
+
+# Manual:
+npm install astro@latest
+pnpm upgrade astro --latest
+yarn upgrade astro --latest
+```
+
+## Experimental: Server Islands
+
+Today we're announcing experimental support for **Server Islands**, bringing islands architecture to the server. With Server Islands you mark components to run dynamically on the server during page load, allowing the rest of the page to be cached at the edge.
+
+In 2021 Astro brought the idea of [islands architecture](https://astro.build/blog/introducing-astro/) to the mainstream. We noticed that developers were building content oriented sites using frontend frameworks even though little of the page were dynamic. Islands architecture provides a way to limit the amount of JavaScript loaded in the client without sacrificing the DX (developer experience) that modern frameworks afford.
+
+With **Server Islands** we are extending this same architecture to the server side as well. Within any given web page you might have content that is:
```

Comment:
"With **Server Islands** we are extending this same architecture to the server side as well." The phrase works well here, but also more reason to not use it in the first paragraph if it exists here

Reply:
This part of the post flows funny.

- Sentence 1: Alright, lets get into what Server Islands are
- Sentence 2: Wait, lets go back to talking about your website again

My advice would be to get all of the "state of the web / Astro / your website today" stuff out of the way, and then get into "what are Server Islands". So in that case, cut _"With Server Islands we are extending this same architecture to the server side as well."_ and move it to after the diagram. 

Also, that makes it suddenly more clear that you're spending a lot of time talking about "the way of the world" vs. "server islands" which is what people are really are here to hear an update on. Can you trim down the background content to 1-2 paragraphs + that diagram, and then get to the good stuff about what Server Islands are?